app/controllers/users_controller.rb

```diff
@@ -16,10 +25,52 @@ def create
     end
   end
 
+  def edit
+  end
+
+  def update
+    if @user.update_attributes(user_params)
+      redirect_to @user, notice: 'Profile updated'
+    else
+      render 'edit'
+    end
+  end
+
+  def destroy
+    find_user.destroy
+    flash[:success] = "User deleted"
+    redirect_to users_url
+  end
+
+
   private
 
+  def find_user
+    User.find(params[:id])
+  end
+
   def user_params
     params.require(:user).permit(:name, :email, :password,
                                  :password_confirmation)
   end
+
+  def logged_in_user
+    store_location
+    unless logged_in?
+      flash[:danger] = "Please log in."
+      redirect_to login_url
+    end
+  end
+
+  def correct_user
```

Comment:
Add this method to the `before_action` and this is a question method, please have `?` at the end of the method definition

Reply:
It is in the `before action`